R/summarize.R

```diff
@@ -1,42 +1,75 @@
+#'   column indicated by \code{interp.format}.
+#' @param fitdat data frame record of constituent+discharge measurements, for
+#'   fitting a model
+#' @param estdat data frame record of discharge measurements, for making
+#'   predictions from a model
+#' @importFrom dplyr mutate bind_cols select %>% everything
+#' @examples
+#' data(lamprey_nitrate)
+#' data(lamprey_discharge)
+#' md <- metadata(constituent="NO3", flow="DISCHARGE", 
+#'   dates="DATE", conc.units="mg L^-1", flow.units="cfs", load.units="kg", 
+#'   load.rate.units="kg d^-1", site.name="Lamprey River, NH",
+#'   basin.area=50, flow.basin.area=65, basin.area.units='ha',
+#'   site.id="1073500", custom=list(data.source="USGS NWIS, waterdata.usgs.gov"))
+#' sitesum <- summarizeInputs(metadata=md, fitdat=lamprey_nitrate[,1:3], 
+#'   estdat=lamprey_discharge)
+summarizeInputs <- function(metadata, fitdat, estdat) {
   
-  #add metrics that have to be calculated
-  if(!is.Date(constit.df$date)) {
-    constit.df <- mutate(constit.df, date = as.Date(date))
-  }
+  constituent <- flow <- load.rate <- dates <- station <- site.name <- 
+    site.id <- consti.name <- flow.basin.area <- basin.area <- '.dplyr.var'
+  
+  # convert metadata into data.frame and add a statistic or two
+  site.info <- 
+    as.data.frame(metadata) %>%
+    select(-constituent, -flow, -load.rate, -dates, -station) %>%
+    select(site.name, site.id, consti.name, everything()) %>%
+    mutate(basin.area.ratio.QC = flow.basin.area / basin.area)
   
-  #will need work if summarizeSites will deal with multiple sites
-  #could be useful as a stand-alone function?
-  dateStats <- getDateStats(constit.df$date)
+  # compute date statistcs for both input datasets
+  fitdat.stats <- summarizeInput(metadata, fitdat)
+  estdat.stats <- summarizeInput(metadata, estdat)
   
-  site.info <- bind_cols(site.info, dateStats)
+  # combine all info into a single data.frame row
+  all.info <- data.frame(
+    site.info, 
+    fitdat=fitdat.stats, 
+    estdat=estdat.stats, 
+    stringsAsFactors=FALSE)
   
-  return(site.info)
+  return(all.info)
 }
 
-#' Get various summary statistics on a date vector
-#'
-#' @param date.col vector of dates
+#' Get summary statistics for a single input data.frame
 #' 
-#' @return data frame of date statistics, including 
-#'
-getDateStats <- function(date.col) {
-  #is there a function that could be used instead of this?
-  start <- min(date.col)
-  end <- max(date.col)
-  n <- length(date.col)
-  statDF <- data.frame(start, end, n, stringsAsFactors = FALSE)
-  return(statDF)
+#' @param metadata object of class metadata, describing the site and data
+#' @param data data.frame of input data, either for model fitting (concentration
+#'   and discharge) or prediction (discharge only)
+#' @return data frame of statistics about the input data
+#' @importFrom stats median
+#' @keywords internal
+summarizeInput <- function(metadata, data) {
```

Comment:
Could we rename this function to something more specific about dates or time?  It is a bit confusing to have both `summarizeInputs` and `summarizeInput`